src/middlewared/middlewared/plugins/zfs.py

```diff
@@ -424,8 +424,9 @@ class Config:
     def query(self, filters=None, options=None):
         # If we are only filtering by name, pool and type we can use
         # zfs(8) which is much faster than py-libzfs
+        options = options or {}
```

Comment:
Is this really necessary? options should default to `{}`

Reply:
`options` by default is `None` as set by `query-options` schema

Reply:
Please elaborate, every Dict is special and should default to {}

Reply:
I see, you said query-options, not in this query. I think we should remove default=None there

Reply:
That makes more sense ;)